### PR TITLE
Simplify multi head attention

### PR DESCRIPTION
**Pull request by coder1729**
I observed that certain things like linear_keys, shape(..) are scattered throughout the multi headed attention file.

### CODE REVIEW

File: onmt/modules/multi_headed_attn.py [hunk 1]
```diff
@@ -120,14 +120,16 @@ def unshape(x):
             return x.transpose(1, 2).contiguous() \
                     .view(batch_size, -1, head_count * dim_per_head)
 
+        query, key, value = self.linear_query(query), \
+                            self.linear_keys(key), \
+                            self.linear_values(value)
+        key = shape(key)
```
Comment (vince62s) [inline onmt/modules/multi_headed_attn.py]: it was 
self.linear_keys(query),\	
self.linear_values(query)

File: onmt/modules/multi_headed_attn.py [hunk 2]
```diff
@@ -189,4 +178,4 @@ def unshape(x):
                   query_len, key_len)[:, 0, :, :] \
             .contiguous()
 
-        return output, top_attn
+        return output, top_attn
```
Comment (vince62s) [inline onmt/modules/multi_headed_attn.py]: you got to fix flake but I think you're change is not good anyway, see my other comment.

File: onmt/modules/multi_headed_attn.py (hunk 1 above)
Comment (ishanch) [reply onmt/modules/multi_headed_attn.py]: in self attention, key,value and query are same right? So it is equivalent

Comment (vince62s) [reply onmt/modules/multi_headed_attn.py]: You are correct.

**Comment by vince62s** [comment]
you have more flakes to fix.
I'll ask @francoishernandez  to review if he's fine with these changes, which makes things simpler but less easy to read IMO.

**Pull request closed without merging**